### ISSUE TITLE
Panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by AlexLab** [thecsw/mira#7]
Hi!

I´m getting this error running this code example. 
```
package main

import (
	"fmt"

	"github.com/thecsw/mira"
)

func main() {
	x := mira.Credentials{ClientId: "*", ClientSecret: "*", Username: "*", Password: "*", UserAgent: "*"}
	r, _ := mira.Init(x)
	sort := "top"
	var limit int = 25
	duration := "all"
	subs, _ := r.Subreddit("all").Submissions(sort, duration, limit)
	for _, v := range subs {
		fmt.Println("Submission Title: ", v.GetTitle())
	}
}
```
Error:
```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0xffffffff addr=0x0 pc=0x2e358e]

goroutine 1 [running]:
github.com/thecsw/mira.(*Reddit).addQueue(...)
	/tmp/gopath892572453/pkg/mod/github.com/thecsw/mira@v3.5.1+incompatible/reddit.go:624
github.com/thecsw/mira.(*Reddit).Subreddit(...)
	/tmp/gopath892572453/pkg/mod/github.com/thecsw/mira@v3.5.1+incompatible/reddit.go:50
main.main()
	/tmp/sandbox199244126/prog.go:15 +0x10e
```

Sure i placed my login info right but getting the same error on 2 of my devices and on play.golang.org.
I´m definitely not an go expret so don´t know if there is an easy fix?

**Comment by thecsw** [comment]
@AlexLab I believe your Reddit login errored out and returned `nil` for the `*mira.Reddit` instance! In your code, try to do some checking for the error

``` go
r, err := mira.Init(x)
if err != nil {
	fmt.Error(err)
}
```

The error will have the reddit returned error!

**Comment by AlexLab** [comment]
Ok it works now :) thanks!

Just some additional info: 
I have an older verison of mira and with that it worked without any issue. Recently i got a new pc and with that i downloaded the latest version, copy&pasted my login info aaand error :(. So to make sure there is no issue with my setup i tried it on play.golang.org also i got the same error. I then created a (not only once - several times to make sure) new reddit app in my account - didn´t work either. 

I ended up creating a new reddit acount and now it works. Not sure why.. on my 1st account i have no other apps running and i really didnt burst out a lot of requests (only 1 every hour or so)
